src/components/timeline/Markers.js

```diff
@@ -27,6 +28,196 @@ export const MIN_MARKER_WIDTH = 0.3;
+    } = this.props;
+
+    if (height === 0 || width === 0) {
+      // bail out early if the size isn't known yet.
+      return;
+    }
+
+    const devicePixelRatio = c.ownerDocument
+      ? c.ownerDocument.defaultView.devicePixelRatio
+      : 1;
+    const pixelWidth = Math.round(width * devicePixelRatio);
+    const pixelHeight = Math.round(height * devicePixelRatio);
+
+    if (c.width !== pixelWidth || c.height !== pixelHeight) {
+      c.width = pixelWidth;
+      c.height = pixelHeight;
+    }
+    const ctx = c.getContext('2d');
+    if (ctx === null || ctx === undefined) {
+      return;
+    }
+
+    ctx.clearRect(0, 0, pixelWidth, pixelHeight);
+    ctx.scale(devicePixelRatio, devicePixelRatio);
+
+    let previousPos = null;
+    for (const markerIndex of markerIndexes) {
+      const marker = getMarker(markerIndex);
+      const { start, dur, name } = marker;
+      let pos = ((start - rangeStart) / (rangeEnd - rangeStart)) * width;
+      pos = Math.round(pos * devicePixelRatio) / devicePixelRatio;
+
+      if (previousPos === pos && dur === 0) {
+        // This position has already been drawn, let's move to the next marker!
+        continue;
+      }
+      previousPos = pos;
+      const itemWidth = Number.isFinite(dur)
+        ? Math.max(
+            (dur / (rangeEnd - rangeStart)) * width,
+            MIN_MARKER_WIDTH / devicePixelRatio
+          )
+        : Number.MAX_SAFE_INTEGER;
+      const markerStyle =
+        name in markerStyles ? markerStyles[name] : markerStyles.default;
+      ctx.fillStyle = markerStyle.background;
+      if (markerStyle.squareCorners) {
+        ctx.fillRect(pos, markerStyle.top, itemWidth, markerStyle.height);
+      } else {
+        _drawRoundedRect(
+          ctx,
+          pos,
+          markerStyle.top,
+          itemWidth,
+          markerStyle.height,
+          1 / devicePixelRatio
+        );
+      }
+      if (markerStyle.borderLeft !== null) {
+        ctx.fillStyle = markerStyle.borderLeft;
+        ctx.fillRect(pos, markerStyle.top, 1, markerStyle.height);
+      }
+      if (markerStyle.borderRight !== null) {
+        ctx.fillStyle = markerStyle.borderRight;
+        ctx.fillRect(
+          pos + itemWidth - 1,
+          markerStyle.top,
+          1,
+          markerStyle.height
+        );
+      }
+      const markerState = this._getMarkerState(marker);
+      if (markerState === 'HOVERED' || markerState === 'PRESSED') {
+        ctx.fillStyle = overlayFills[markerState];
+        if (markerStyle.squareCorners) {
+          ctx.fillRect(pos, markerStyle.top, itemWidth, markerStyle.height);
+        } else {
+          _drawRoundedRect(
+            ctx,
+            pos,
+            markerStyle.top,
+            itemWidth,
+            markerStyle.height,
+            1 / devicePixelRatio
+          );
+        }
+      }
+    }
+    ctx.scale(1 / devicePixelRatio, 1 / devicePixelRatio);
+  }
+
+  _scheduleDraw() {
+    if (!this._requestedAnimationFrame) {
```

Comment:
as explained above, I changed the previous `scheduleDraw` to align with practices in the other components.